File: javaleet/P28.java
public class P28 {
    public int strStr(String haystack, String needle) {
        for(int i = 0; i< haystack.length(); i++ ){
            if (haystack[i] == needle[0]) {
                for(int j=0; j<needle.length(); j++){
                    if(haystack[i+j] != needle[j]) {
                        break;
                    }
                }
            }
        }
    }   
    public static void main(String[] args) {
        P28 myP = new P28();
        System.out.println(myP.strStr("hello", "ll"));
        System.out.println(myP.strStr("aaaaa", "bba"));
        System.out.println(myP.strStr("", ""));
    }
}

File: p28.py
class Solution:
    def strStr(self, haystack, needle):
        return haystack.find(needle)

    def strStr1(self, haystack, needle):
        print(len(haystack))
        print(len(needle))
        if needle == "":
            return 0

        if haystack == "":
            return -1
        if len(haystack) < len(needle):
            return -1
        for idx in range(len(haystack)):
            if haystack[idx] == needle[0]:
                if haystack[idx:len(needle)+idx] == needle:
                    return idx
        return -1

if __name__ == "__main__":
    s = Solution()
    print(s.strStr("hello", "ll"))
    print(s.strStr("aaaaa", "baa"))
